### PR TITLE
Added graphic modes

### PR DESCRIPTION
**Pull request by nikp123**
I added x and sdl as graphics modes for the CAVA visualizer. I would like if you could merge these projects.
This may be a replacment for your OpenGL idea.

### CODE REVIEW

**Comment by nikp123** [comment]
Sorry, accidents happen.

File: configure.ac [hunk 1]
```diff
@@ -36,6 +36,35 @@ AC_CHECK_HEADERS([pthread.h],
...
+dnl ######################
+
+AC_CHECK_LIB(X11, XOpenDisplay, have_xlib=yes, have_xlib=no)
+
+if [[ $have_xlib = "yes" ]] ; then
+	LIBS="$LIBS -lX11"
+	CPPFLAGS="$CPPFLAGS -DXLIB"
+fi
+
+if [[ $have_xlib= "no" ]] ; then
```
Comment (benwaffle) [inline configure.ac]: missing a space before `=`

**Comment by nikp123** [comment]
@benwaffle Thanks for noticing that typo ;)

**Comment by karlstav** [comment]
hi @nikp123 I just made a new branch called `gui`, if you make a pull request to that branch I can pull it and we can work on it from there.

**Comment by nikp123** [comment]
@karlstav changed it ;)

**Comment by karlstav** [comment]
oh shit git, should have created the new branch before merging the new more color branch. now we have conflicts... must look at this later

**Comment by nikp123** [comment]
@karlstav WAIT, did you accidentally pull to the wrong branch. Or I did something wrong!?

**Comment by nikp123** [comment]
@karlstav Removed conflicts

**Comment by benwaffle** [comment]
aww, it couldn't be fixed? SDL2 should work on wayland too ...

**Comment by benwaffle** [comment]
It crashes when i hit `c` with X11 output, at cava.c:1209, `free(color);`

**Comment by nikp123** [comment]
@benwaffle I'll get that checked in a few hours or maybe days. I don't have much free time now, sorry :(

**Comment by nikp123** [comment]
@benwaffle It probably is unallocating a constant (which is impossible, btw.). Setting both colors to hex value instead of a "color name" would fix that issue.

**Pull request merged**